### PR TITLE
Support Routing Stub Tests for Javascript Driver

### PR DESCRIPTION
**Pull request by bigmontz**
Enable the tests and make the necessary adjustments on it.

### CODE REVIEW

File: tests/stub/routing.py [hunk 1]
```diff
@@ -125,7 +135,7 @@ def get_db(self):
     # Checks that routing is used to connect to correct server and that parameters for
     # session run is passed on to the target server (not the router).
     def test_session_run_read(self):
-        if get_driver_name() not in ['go', 'dotnet', 'java']:
+        if get_driver_name() not in ['go', 'dotnet', 'java', 'javascript']:
```
Comment (2hdddg) [inline tests/stub/routing.py]: We can remove this one

File: tests/stub/routing.py [hunk 2]
```diff
@@ -136,11 +146,11 @@ def test_session_run_read(self):
         session.close()
         driver.close()
         self._routingServer.done()
-        self._readServer.done()
+        self._readServer.done() 
 
     # Same test as for session.run but for transaction run.
     def test_tx_run_read(self):
-        if get_driver_name() not in ['go', 'dotnet', 'java']:
+        if get_driver_name() not in ['go', 'dotnet', 'java', 'javascript']:
```
Comment (2hdddg) [inline tests/stub/routing.py]: Remove

File: tests/stub/routing.py [hunk 3]
```diff
@@ -157,7 +167,7 @@ def test_tx_run_read(self):
 
     # Checks that write server is used
     def test_session_run_write(self):
-        if get_driver_name() not in ['go', 'dotnet', 'java']:
+        if get_driver_name() not in ['go', 'dotnet', 'java', 'javascript']:
```
Comment (2hdddg) [inline tests/stub/routing.py]: Remove

File: tests/stub/routing.py [hunk 4]
```diff
@@ -172,7 +182,7 @@ def test_session_run_write(self):
 
     # Checks that write server is used
     def test_tx_run_write(self):
-        if get_driver_name() not in ['go', 'dotnet', 'java']:
+        if get_driver_name() not in ['go', 'dotnet', 'java', 'javascript']:
```
Comment (2hdddg) [inline tests/stub/routing.py]: Remove

File: tests/stub/routing.py [hunk 5]
```diff
@@ -115,8 +119,14 @@ def get_vars(self):
...
             "#EXTRA_HELLO_PROPS#": get_extra_hello_props(),
         }
-        if get_driver_name() in ['dotnet', 'java']:
+        v["#HELLO_ROUTINGCTX#"] = v["#ROUTINGCTX#"]
+
+        if get_driver_name() in ['dotnet', 'java', 'javascript']:
             v["#DBPARAM#"] = "database"
+
+        if get_driver_name() in ['javascript']:
+            v["#HELLO_ROUTINGCTX#"] = '{"region": "china", "policy": "my_policy"}'
```
Comment (2hdddg) [inline tests/stub/routing.py:128]: Not related to the PR, but why is this only needed on js driver?

File: tests/stub/routing.py [hunk 6]
```diff
@@ -313,14 +329,15 @@ def get_vars(self):
...
+            "#ROUTING#": '", routing": null' if get_driver_name() not in ['javascript'] else ''
         }
 
 
     # Checks that routing is disabled when URI is bolt, no routing in HELLO and
     # no call to retrieve routing table. From bolt >= 4.1 the routing context is
     # used to disable/enable server side routing.
     def test_session_run_read(self):
-        if get_driver_name() not in ['go', 'dotnet', 'java']:
+        if get_driver_name() not in ['go', 'dotnet', 'java', 'javascript']:
```
Comment (2hdddg) [inline tests/stub/routing.py]: Remove

File: tests/stub/routing.py [hunk 7]
```diff
@@ -275,10 +285,16 @@ def get_vars(self):
...
         }
+        
         if get_driver_name() in ['go']:
             v["#ROUTINGMODE#"] = '"mode": "r"'
+        
         if get_driver_name() in ['java']:
             v["#EXTR_HELLO_ROUTING_PROPS#"] = ', "routing": ' + v['#ROUTINGCTX#'] 
+        
+        if get_driver_name() in ['javascript']:
+            v["#ROUTINGCTX#"] = '{"region": "china", "policy": "my_policy"}'
```
Comment (2hdddg) [inline tests/stub/routing.py:284]: Interesting, why is this only in js driver?

File: tests/stub/routing.py (hunk 1 above)
Comment (bigmontz) [reply tests/stub/routing.py]: python is not supported by testkit?

File: tests/stub/routing.py (hunk 5 above)
Comment (bigmontz) [reply tests/stub/routing.py:128]: It's a strange behaviour. 

On the 4.x, it sends the context with the `address` when it call the procedure,  but it sends the routing context without the `address` to the HELLO message. 

On 3, it doesn't send the `address` to the procedure and the java driver is the only one to send routing context on the HELLO. 

I think the lack of address on the routing message for the 3.x could be a bug/miss-behaviour. But I don't know the impact of this.

File: tests/stub/routing.py (hunk 1 above)
Comment (2hdddg) [reply tests/stub/routing.py]: No backend yet...

**Pull request merged**